u_common.go

```diff
@@ -74,49 +74,68 @@ const (
 	helloRandomizedNoALPN
 )
 
+type ClientHelloSpec struct {
+	ClientRandom       []byte
```

Comment:
Can this be nil?

Reply:
Yes, but now that I thought of it, we probably shouldn't have it in spec at all, as it's supposed to be a generic descriptor of configuration.
Will remove.